lib/tasks/convert_application_yml_to_proper_types.rake

```diff
@@ -0,0 +1,64 @@
+namespace :convert_application_yml_to_proper_types do
+  desc 'Take application.yml input file and convert to proper type'
+  task :convert, [:filename] => :environment do |t, args|
+    VALUE_TYPES = ['float', 'string', 'boolean', 'json', 'timestamp', 'integer']
+    filename = args[:filename]
+
+    if filename
+      content = ''
+      add_spaces = false
+      File.open(filename+'.tmp', 'w') do |newfile|
+        IO.foreach(filename) do |l|
+          add_spaces = true if l.include?('development:')
+          content=l
+          key_value = l.split(':')
+
+          if key_value.length == 2 && !key_value[1].strip.blank?
+            key = key_value[0].strip
+            value = key_value[1].strip
```

Comment:
a few ideas:
1. nit, don't use a single-letter variable name, especially if we're just going to reassign it later
1. nit, add spacing around the assignment
2. use array destructuring to help give the variables their names

```suggestion
        IO.foreach(filename) do |line|
          add_spaces = true if line.include?('development:') # see comment about checking other envs as well
          key, value, *rest = line.split(':')
          key&.strip!
          value&.strip!

          if key.present? && value.present? 
```

Reply:
All taken care of